test/e2e/case18_compliance_api_test.go

```diff
@@ -112,4 +110,757 @@ var _ = Describe("Test policy webhook", Label("compliance-events-api"), Ordered,
 			}, defaultTimeoutSeconds, 1).Should(Succeed())
 		})
 	})
+
+	Describe("Test POSTing Events", func() {
+		Describe("POST one valid event with including all the optional fields", func() {
+			payload := []byte(`{
+				"cluster": {
+					"name": "cluster1",
+					"cluster_id": "test1-cluster1-fake-uuid-1"
+				},
+				"parent_policy": {
+					"name": "policies.etcd-encryption1",
+					"categories": ["cat-1", "cat-2"],
+					"controls": ["ctrl-1"],
+					"standards": ["stand-1"]
+				},
+				"policy": {
+					"apiGroup": "policy.open-cluster-management.io",
+					"kind": "ConfigurationPolicy",
+					"name": "etcd-encryption1",
+					"namespace": "local-cluster",
+					"spec": "{\"test\":\"one\",\"severity\":\"low\"}"
+				},
+				"event": {
+					"compliance": "NonCompliant",
+					"message": "configmaps [etcd] not found in namespace default",
+					"timestamp": "2023-01-01T01:01:01.111Z",
+					"metadata": {"test": true},
+					"reported_by": "optional-test"
+				}
+			}`)
+
+			BeforeAll(func(ctx context.Context) {
+				By("POST the event")
+				Eventually(postEvent(ctx, payload), "5s", "1s").ShouldNot(HaveOccurred())
+			})
+
+			It("Should have created the cluster in a table", func() {
+				rows, err := db.Query("SELECT * FROM clusters WHERE cluster_id = $1", "test1-cluster1-fake-uuid-1")
+				Expect(err).ToNot(HaveOccurred())
+
+				count := 0
+				for rows.Next() {
+					var (
+						id        int
+						name      string
+						clusterID string
+					)
+					err := rows.Scan(&id, &name, &clusterID)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(id).NotTo(Equal(0))
+					Expect(name).To(Equal("cluster1"))
+					count++
+				}
+
+				Expect(count).To(Equal(1))
+			})
+
+			It("Should have created the parent policy in a table", func() {
+				rows, err := db.Query("SELECT * FROM parent_policies WHERE name = $1", "policies.etcd-encryption1")
+				Expect(err).ToNot(HaveOccurred())
+
+				count := 0
+				for rows.Next() {
+					var (
+						id     int
+						name   string
+						cats   pq.StringArray
+						ctrls  pq.StringArray
+						stands pq.StringArray
+					)
+
+					err := rows.Scan(&id, &name, &cats, &ctrls, &stands)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(id).NotTo(Equal(0))
+					Expect(cats).To(ContainElements("cat-1", "cat-2"))
+					Expect(ctrls).To(ContainElements("ctrl-1"))
+					Expect(stands).To(ContainElements("stand-1"))
+					count++
+				}
+
+				Expect(count).To(Equal(1))
+			})
+
+			It("Should have created the policy in a table", func() {
+				rows, err := db.Query("SELECT * FROM policies WHERE name = $1", "etcd-encryption1")
+				Expect(err).ToNot(HaveOccurred())
+
+				count := 0
+				for rows.Next() {
+					var (
+						id       int
+						kind     string
+						apiGroup string
+						name     string
+						ns       *string
+						pid      *int
+						spec     *string
+						specHash *string
+						severity *string
```

Comment:
So you have to use a pointer for potentially null values?

Reply:
Confirmed (and maybe saved you an experiment): if a column is NULL and you don't try to put it into a pointer, you get an error like this:
```
sql: Scan error on column index 8, name \"severity\": converting NULL to string is unsupported
```

On the other hand, I think you can always use a pointer - as far as I can tell, I could use a pointer for any of the `id` fields. It's just slightly annoying to de-reference later, and makes me worry about nil pointers.